opencti-platform/opencti-front/src/private/components/workspaces/investigations/Dialog/InvestigationRollBackExpandDialog.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import { useFormatter } from 'src/components/i18n';
+
+type InvestigationRollBackExpandDialogProps = {
+  closeDialog: () => void;
+  handleRollBackToPreExpansionState: () => void;
+  isOpen: boolean;
+};
+
+const InvestigationRollBackExpandDialog = ({ closeDialog, handleRollBackToPreExpansionState, isOpen }: InvestigationRollBackExpandDialogProps) => {
+  const { t_i18n, fldt } = useFormatter();
+
+  const handleSubmit = () => {
+    handleRollBackToPreExpansionState();
+    closeDialog();
+  };
+
+  const getLastRollBackExpandDate = () => {
+    const storedPreExpansion = sessionStorage.getItem('preExpansionStateList');
```

Comment:
maybe we could have a dedicated file with functions to store & retrieve investigation state in session storage ? it would keep the logic in one place. @labo-flg @ValentinBouzinFiligran